<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>UserInput_Dlg</class>
 <widget class="QWidget" name="UserInput_Dlg">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>435</width>
    <height>355</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>4</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>User Input</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Select the type of user input device to query!</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Mouse</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <widget class="QComboBox" name="comboBox_mouseType"/>
       </item>
       <item row="1" column="0">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>352</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Keyboard</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Key:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="KeyboardListenerLineEdit" name="lineEdit_keyCapture">
           <property name="focusPolicy">
            <enum>Qt::StrongFocus</enum>
           </property>
           <property name="text">
            <string>Press the key to capture</string>
           </property>
           <property name="echoMode">
            <enum>QLineEdit::Normal</enum>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QRadioButton" name="radioButton_vKey">
         <property name="focusPolicy">
          <enum>Qt::StrongFocus</enum>
         </property>
         <property name="text">
          <string>Use Virtual Key Codes</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">buttonGroup_kt</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radioButton_scanCodes">
         <property name="text">
          <string>Use Keyboard Scan Codes</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">buttonGroup_kt</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radioButton_anyKey">
         <property name="text">
          <string>Any Key</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">buttonGroup_kt</string>
         </attribute>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>350</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_4">
      <attribute name="title">
       <string>Joystick</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Joystick nr:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="cbJoystickNr">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Command:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QComboBox" name="cbJoystickCmd">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>111</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_5">
      <attribute name="title">
       <string>Touch</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QComboBox" name="comboBox_touchType"/>
       </item>
       <item>
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>28</width>
           <height>139</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Button Mode</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QRadioButton" name="radioButton_bm0">
        <property name="text">
         <string>1 on Unpressed</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">buttonGroup_bm</string>
        </attribute>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="radioButton_bm1">
        <property name="text">
         <string>1 on Pressing</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">buttonGroup_bm</string>
        </attribute>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="radioButton_bm2">
        <property name="text">
         <string>1 on Pressed</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">buttonGroup_bm</string>
        </attribute>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="radioButton_bm3">
        <property name="text">
         <string>1 on Releasing</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">buttonGroup_bm</string>
        </attribute>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KeyboardListenerLineEdit</class>
   <extends>QLineEdit</extends>
   <header>KeyboardListenerLineEdit.h</header>
   <slots>
    <signal>keyPressed(QKeyEvent*)</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>lineEdit_keyCapture</sender>
   <signal>keyPressed(QKeyEvent*)</signal>
   <receiver>UserInput_Dlg</receiver>
   <slot>keyListenerPressed(QKeyEvent*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>119</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>212</x>
     <y>221</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonGroup_kt</sender>
   <signal>idClicked(int)</signal>
   <receiver>UserInput_Dlg</receiver>
   <slot>keyCodeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>212</x>
     <y>221</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tabWidget</sender>
   <signal>currentChanged(int)</signal>
   <receiver>UserInput_Dlg</receiver>
   <slot>deviceTypeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>221</x>
     <y>220</y>
    </hint>
    <hint type="destinationlabel">
     <x>212</x>
     <y>221</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_mouseType</sender>
   <signal>activated(int)</signal>
   <receiver>UserInput_Dlg</receiver>
   <slot>mouseTypeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>101</x>
     <y>45</y>
    </hint>
    <hint type="destinationlabel">
     <x>212</x>
     <y>221</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonGroup_bm</sender>
   <signal>idClicked(int)</signal>
   <receiver>UserInput_Dlg</receiver>
   <slot>buttonModeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>217</x>
     <y>177</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbJoystickNr</sender>
   <signal>activated(int)</signal>
   <receiver>UserInput_Dlg</receiver>
   <slot>joystickChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>119</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>217</x>
     <y>177</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbJoystickCmd</sender>
   <signal>activated(int)</signal>
   <receiver>UserInput_Dlg</receiver>
   <slot>joystickChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>119</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>217</x>
     <y>177</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_touchType</sender>
   <signal>activated(int)</signal>
   <receiver>UserInput_Dlg</receiver>
   <slot>touchChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>326</x>
     <y>51</y>
    </hint>
    <hint type="destinationlabel">
     <x>435</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>keyListenerPressed(QKeyEvent*)</slot>
  <slot>keyCodeChanged()</slot>
  <slot>deviceTypeChanged()</slot>
  <slot>mouseTypeChanged()</slot>
  <slot>buttonModeChanged()</slot>
  <slot>joystickChanged()</slot>
  <slot>touchChanged()</slot>
 </slots>
 <buttongroups>
  <buttongroup name="buttonGroup_kt"/>
  <buttongroup name="buttonGroup_bm"/>
 </buttongroups>
</ui>
